mail_notification_custom_subject/tests/test_mail_notification_custom_subject.py

```diff
@@ -0,0 +1,167 @@
+            }
+        )
+        # Send message in partner
+        mail_message_1 = self.partner_1.message_post(
+            body="Test", subtype="mail.mt_comment"
+        )
+        # Get message and check subject
+        self.assertEquals(
+            mail_message_1.subject, "Test partner 1 and something different"
+        )
+        self.env["mail.message.custom.subject"].create(
+            {
+                "name": "Test template 3",
+                "model_id": self.env.ref("base.model_res_partner").id,
+                "subtype_ids": [(6, 0, [self.env.ref("mail.mt_comment").id])],
+                "subject_template": "${' and yet something else'}",
+                "position": "append_after",
+            }
+        )
+        # Send message in partner
+        mail_message_2 = self.partner_1.message_post(
+            body="Test", subtype="mail.mt_comment"
+        )
+        # Get message and check subject
+        self.assertEquals(
+            mail_message_2.subject,
+            "Test partner 1 and something different and yet something else",
+        )
+        self.env["mail.message.custom.subject"].create(
+            {
+                "name": "Test template 4",
+                "model_id": self.env.ref("base.model_res_partner").id,
+                "subtype_ids": [(6, 0, [self.env.ref("mail.mt_comment").id])],
+                "subject_template": "${'Re: '}",
+                "position": "append_before",
+            }
+        )
+        # Send message in partner
+        mail_message_3 = self.partner_1.message_post(
+            body="Test", subtype="mail.mt_comment"
+        )
+        # Get message and check subject
+        self.assertEquals(
+            mail_message_3.subject,
+            "Re: Test partner 1 and something different and yet something else",
+        )
+
+    def test_email_subject_template_w_original(self):
+        self.env["mail.message.custom.subject"].create(
+            {
+                "name": "Test template 1",
+                "model_id": self.env.ref("base.model_res_partner").id,
+                "subtype_ids": [(6, 0, [self.env.ref("mail.mt_comment").id])],
+                "subject_template": "${' and something more'}",
+                "position": "append_after",
+            }
+        )
+        # Send message in partner
+        mail_message_1 = self.partner_1.message_post(
+            body="Test", subtype="mail.mt_comment", subject="Test",
+        )
+        # Get message and check subject
+        self.assertEquals(mail_message_1.subject, "Test and something more")
+
+    def test_bad_template_does_not_break(self):
+        self.env["mail.message.custom.subject"].create(
+            {
+                "name": "Test bad template 1",
+                "model_id": self.env.ref("base.model_res_partner").id,
+                "subtype_ids": [(6, 0, [self.env.ref("mail.mt_comment").id])],
+                "subject_template": "${obaject.number_a} and something",
```

Comment:
```suggestion
                "subject_template": "${object.number_a} and something",
```

Reply:
This is intended @carlosdauden 
The tests asserts that a badly formed template will not render, but also won't block the sending of emails.

Reply:
You're right, sorry...